extensions/src/org/pentaho/platform/web/http/filters/PentahoWebContextFilter.java

```diff
@@ -369,6 +368,20 @@ private void printResourcesForContext( String contextName, OutputStream out, Htt
 
   }
 
+  public void printRequireJsCfgStart( OutputStream out ) throws IOException {
+
+    int waitTime;
+
+    try {
```

Comment:
It caches the whole pentaho.xml. Every time you read the setting, it's traversing the XML to get the discrete setting. So yes - it pulls the xml file out of the cache, and then parses it to get the discrete value. My preference is for this method to extract the discrete value from the cache if it's there, or read the setting if it's not.